samples/subsys/smf/smf_calculator/src/smf_calculator_thread.c

```diff
@@ -0,0 +1,700 @@
+		smf_set_handled(&s->ctx);
+		break;
+	default:
+		/* Let parent state handle it: shuts up compiler warning */
+		break;
+	}
+}
+
+static void negated_2_entry(void *obj)
+{
+	LOG_DBG("%s", __func__);
+	struct s_object *s = (struct s_object *)obj;
+
+	negate(&s->operand_2);
+}
+
+static void negated_2_run(void *obj)
+{
+	LOG_DBG("%s", __func__);
+	struct s_object *s = (struct s_object *)obj;
+
+	switch (s->event.event_id) {
+	case DECIMAL_POINT:
+		insert(&s->operand_2, s->event.operand);
+		smf_set_state(&s->ctx, &calculator_states[STATE_FRAC_2]);
+		break;
+	case DIGIT_1_9:
+		insert(&s->operand_2, s->event.operand);
+		smf_set_state(&s->ctx, &calculator_states[STATE_INT_2]);
+		break;
+	case DIGIT_0:
+		/* Don't insert the leading zero */
+		smf_set_state(&s->ctx, &calculator_states[STATE_ZERO_2]);
+		break;
+	case OPERATOR:
+		/* We only care about ignoring the negative key */
+		if (s->event.operand == '-') {
+			smf_set_handled(&s->ctx);
+		}
+		break;
+	case CANCEL_ENTRY:
+		setup_operand(&s->operand_2);
+		smf_set_state(&s->ctx, &calculator_states[STATE_OP_ENTERED]);
+		break;
+	default:
+		/* Let parent state handle it: shuts up compiler warning */
+		break;
+	}
+}
+
+static void operand_2_entry(void *obj)
+{
+	LOG_DBG("%s", __func__);
+	set_display_mode(DISPLAY_OPERAND_2);
+}
+
+static void operand_2_run(void *obj)
+{
+	LOG_DBG("%s", __func__);
+	struct s_object *s = (struct s_object *)obj;
+
+	switch (s->event.event_id) {
+	case CANCEL_ENTRY:
+		setup_operand(&s->operand_2);
+		smf_set_state(&s->ctx, &calculator_states[STATE_OP_ENTERED]);
+		break;
+	case OPERATOR:
+		if (calculate_result(s) == 0) {
+			/* move result into operand_1 to allow chaining */
+			chain_calculations(s);
+			/* Copy in new operand */
+			s->operator_btn = s->event.operand;
+			smf_set_state(&s->ctx, &calculator_states[STATE_OP_CHAINED]);
+		} else {
+			smf_set_state(&s->ctx, &calculator_states[STATE_ERROR]);
+		}
+		break;
+	case EQUALS:
+		if (calculate_result(s) == 0) {
+			/* move result into operand_1 to allow chaining */
+			chain_calculations(s);
+			smf_set_state(&s->ctx, &calculator_states[STATE_RESULT]);
+		} else {
+			smf_set_state(&s->ctx, &calculator_states[STATE_ERROR]);
+		}
+		break;
+	default:
+		/* Let parent state handle it: shuts up compiler warning */
+		break;
+	}
+}
+
+static void zero_2_run(void *obj)
+{
+	LOG_DBG("%s", __func__);
+	struct s_object *s = (struct s_object *)obj;
+
+	switch (s->event.event_id) {
+	case DIGIT_0:
+		/* We ignore leading zeroes */
+		smf_set_handled(&s->ctx);
+		break;
+	case DIGIT_1_9:
+		insert(&s->operand_2, s->event.operand);
+		smf_set_state(&s->ctx, &calculator_states[STATE_INT_2]);
+		break;
+	case DECIMAL_POINT:
+		insert(&s->operand_2, s->event.operand);
+		smf_set_state(&s->ctx, &calculator_states[STATE_FRAC_2]);
+		break;
+	default:
+		/* Let parent state handle it: shuts up compiler warning */
+		break;
+	}
+}
+
+static void int_2_run(void *obj)
+{
+	LOG_DBG("%s", __func__);
```

Comment:
Thanks. Will change.

I put them all at the front because they're common to all functions, where declaring the `s_object` pointer is not, but will change that too.